apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,61 +1,63 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
-import { ListItemComponent } from '../list-item/list-item.component';
+import { NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ContentChild,
+  EventEmitter,
+  Input,
+  Output,
+  TemplateRef,
+} from '@angular/core';
+import { CardItem } from '../../model/card.model';
 
 @Component({
   selector: 'app-card',
   template: `
     <div
       class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4"
       [class]="customClass">
-      <img
-        *ngIf="type === CardType.TEACHER"
-        src="assets/img/teacher.png"
-        width="200px" />
-      <img
-        *ngIf="type === CardType.STUDENT"
-        src="assets/img/student.webp"
-        width="200px" />
+      <ng-content select="[img]"></ng-content>
 
       <section>
-        <app-list-item
-          *ngFor="let item of list"
-          [name]="item.firstName"
-          [id]="item.id"
-          [type]="type"></app-list-item>
+        @for (item of list; track item.id) {
+          <ng-container
+            *ngTemplateOutlet="
+              template;
+              context: { $implicit: item }
+            "></ng-container>
+        }
       </section>
 
       <button
         class="rounded-sm border border-blue-500 bg-blue-300 p-2"
-        (click)="addNewItem()">
+        (click)="this.add.emit()">
         Add
       </button>
     </div>
   `,
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [NgIf, NgFor, NgTemplateOutlet],
+  styles: [
+    `
+      :host {
+        display: inline-block;
+      }
+      :host(.bg-light-red) {
+        background-color: rgba(250, 0, 0, 0.1);
+      }
+      :host(.bg-light-green) {
+        background-color: rgba(0, 250, 0, 0.1);
+      }
+      :host(.bg-light-blue) {
+        background-color: rgba(0, 0, 250, 0.1);
+      }
+    `,
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
+  @Input() list: CardItem[] | null = null;
   @Input() customClass = '';
-
-  CardType = CardType;
-
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore,
-  ) {}
-
-  addNewItem() {
-    if (this.type === CardType.TEACHER) {
-      this.teacherStore.addOne(randTeacher());
-    } else if (this.type === CardType.STUDENT) {
-      this.studentStore.addOne(randStudent());
-    }
-  }
+  @ContentChild(TemplateRef) template!: TemplateRef<{ $implicit: CardItem }>;
```

Comment:
Use the TemplateRef is dangerous because if you decide to add a new template later, or someone in your team, you can have weird bug or side effect. 

You should prefer to use a directive to target to right template

Reply:
I have updated it with empty directive. what do you think about it?

```
import { Directive } from '@angular/core';
@Directive({
  selector: '[list-item-template]',
  standalone: true,
})
export class ListItemTemplateDirective {}
```

and add it inside ContentChild directive

```
@ContentChild(ListItemTemplateDirective, { read: TemplateRef })
  template!: TemplateRef<{ $implicit: CardItem }>;
```

Reply:
yes that's perfect